src/System.Private.CoreLib/shared/System/Reflection/TypeDelegator.cs

```diff
@@ -70,24 +71,24 @@ public TypeDelegator(Type delegatingType)
 
         public override MethodInfo[] GetMethods(BindingFlags bindingAttr) => typeImpl.GetMethods(bindingAttr);
 
-        public override FieldInfo GetField(string name, BindingFlags bindingAttr) => typeImpl.GetField(name, bindingAttr);
+        public override FieldInfo? GetField(string name, BindingFlags bindingAttr) => typeImpl.GetField(name, bindingAttr);
         public override FieldInfo[] GetFields(BindingFlags bindingAttr) => typeImpl.GetFields(bindingAttr);
 
         public override Type GetInterface(string name, bool ignoreCase) => typeImpl.GetInterface(name, ignoreCase);
 
         public override Type[] GetInterfaces() => typeImpl.GetInterfaces();
 
-        public override EventInfo GetEvent(string name, BindingFlags bindingAttr) => typeImpl.GetEvent(name, bindingAttr);
+        public override EventInfo? GetEvent(string name, BindingFlags bindingAttr) => typeImpl.GetEvent(name, bindingAttr);
 
         public override EventInfo[] GetEvents() => typeImpl.GetEvents();
 
-        protected override PropertyInfo GetPropertyImpl(string name, BindingFlags bindingAttr, Binder binder,
-                        Type returnType, Type[] types, ParameterModifier[] modifiers)
+        protected override PropertyInfo? GetPropertyImpl(string name, BindingFlags bindingAttr, Binder? binder,
+                        Type? returnType, Type[]? types, ParameterModifier[]? modifiers)
         {
-            if (returnType == null && types == null)
+            if (returnType == null && types == null) // this condition might need refactoring, (returnType == null && types == null) doesn't mean both != null
                 return typeImpl.GetProperty(name, bindingAttr);
-            else
-                return typeImpl.GetProperty(name, bindingAttr, binder, returnType, types, modifiers);
+            else 
+                return typeImpl.GetProperty(name, bindingAttr, binder, returnType, types!, modifiers);
```

Comment:
As commented above we might want to update the condition

Reply:
I don't think we should update the condition here, actually we don't have other thing to do but to use the `!` operator. Types can only be null if `returnType` is null. However if `returnType` is not null, then we will throw an ArgumentNullException if `types` are `null` -- I think this is correct... @stephentoub would an attribute for this case make sense?